hbase-common/src/main/java/org/apache/hadoop/hbase/HConstants.java

```diff
@@ -136,6 +136,12 @@
   /** Default value for the max percent of regions in transition */
   public static final double DEFAULT_HBASE_MASTER_BALANCER_MAX_RIT_PERCENT = 1.0;
 
+  /** Time in milliseconds to wait meta region assignment, when moving non-meta regions. */
+  public static final String HBASE_MASTER_WAITING_META_ASSIGNMENT_TIMEOUT =
+    "hbase.master.waiting.meta.assignment.timeout";
+
+  public static final long HBASE_MASTER_WAITING_META_ASSIGNMENT_TIMEOUT_DEFAULT = 5000;
```

Comment:
Let's keep it 10000 (10 s)?

Reply:
Not sure if 10s is better. Actually, 5s is my intuition. Have changed in the latest commit. But could you explain the reason to change to 10s a little bit?